includes/abstracts/abstract-wc-product.php

```diff
@@ -27,7 +27,7 @@ class WC_Product extends WC_Abstract_Legacy_Product {
 	 * @var array
 	 */
 	protected $data = array(
-		'name'               => '',
+		'name'               => 'New Product',
```

Comment:
Are we sure we want a default name here? If we do, it should be wrapped in translation functions.